packages/loopring_v3/contracts/impl/ProtocolFeeVault.sol

```diff
@@ -25,73 +25,48 @@ import "../lib/MathUint.sol";
 import "../lib/ReentrancyGuard.sol";
 
 import "../iface/IProtocolFeeVault.sol";
-
-
-/// @dev See https://github.com/Loopring/protocols/blob/master/packages/oedax_v1/contracts/iface/IOedax.so
-contract IOedax
-{
-    /// @param askToken The ask (base) token. Prices are in form of 'bids/asks'.
-    /// @param bidToken The bid (quote) token.
-    /// @param minAskAmount The minimum ask amount.
-    /// @param minBidAmount The minimum bid amount.
-    /// @param P Numerator part of the target price `p`.
-    /// @param S Price precision -- (_P / 10**_S) is the float value of the target price.
-    /// @param M Price factor. `p * M` is the maximum price and `p / M` is the minimum price.
-    /// @param T1 The minimum auction duration in second.
-    /// @param T2 The maximum auction duration in second.
-    /// @return auctionAddr Auction address.
-    function createAuction(
-        address askToken,
-        address bidToken,
-        uint    minAskAmount,
-        uint    minBidAmount,
-        uint64  P,
-        uint64  S,
-        uint8   M,
-        uint    T1,
-        uint    T2
-        )
-        public
-        payable
-        returns (address payable auctionAddr);
-
-    mapping (address => uint) public tokenRankMap;
-
-    uint public creatorEtherStake;
-}
-
-
-/// @dev See https://github.com/Loopring/protocols/blob/master/packages/oedax_v1/contracts/iface/IAuction.so
-contract IAuction
-{
-    function settle() public;
-    function ask(uint amount) external returns (uint accepted);
-}
-
+import "../iface/ITokenSeller.sol";
 
 /// @title An Implementation of IProtocolFeeVault.
 /// @author Daniel Wang - <daniel@loopring.org>
 contract ProtocolFeeVault is Claimable, ReentrancyGuard, IProtocolFeeVault
 {
-    uint public constant MIN_ETHER_TO_KEEP = 1 ether;
     using AddressUtil       for address;
     using AddressUtil       for address payable;
     using ERC20SafeTransfer for address;
     using MathUint          for uint;
 
-    constructor(
-        address _lrcAddress,
-        address _userStakingPoolAddress
-        )
+    constructor(address _lrcAddress)
         Claimable()
         public
     {
         require(_lrcAddress != address(0), "ZERO_ADDRESS");
-        require(_userStakingPoolAddress != address(0), "ZERO_ADDRESS");
-
-        allowOwnerWithdrawal = true;
         lrcAddress = _lrcAddress;
+    }
+
+    function updateSettings(
+        address _userStakingPoolAddress,
+        address _tokenSellerAddress,
+        address _daoAddress
```

Comment:
need separate to 3 single update when only update one which can save gas?

Reply:
this method is rarely used, so saving gas  is not a concern.